components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -612,6 +612,16 @@ abstract class AbstractFetchDownloadService : Service() {
     private fun sendDownloadStopped(downloadState: DownloadJobState) {
         downloadState.notifiedStopped = true
 
+        /**
+         * In cases when we don't get the file size provided initially, we have to
+         * use downloadState.currentBytesCopied, as a fallback.
+         */
+        val fileSizeNotFound = downloadState.state.contentLength == null || downloadState.state.contentLength == 0L
+        if (downloadState.status == COMPLETED && fileSizeNotFound) {
+            val newState = downloadState.state.copy(contentLength = downloadState.currentBytesCopied)
+            updateDownloadState(newState)
```

Comment:
@csadilek `updateDownloadState` internally updates the store, there is any way to warranty that this action was performed before executing the code from line 625 to 627

```kotlin
    @VisibleForTesting
    internal fun updateDownloadState(updatedDownload: DownloadState) {
        downloadJobs[updatedDownload.id]?.state = updatedDownload
        store.dispatch(DownloadAction.UpdateQueuedDownloadAction(updatedDownload))
    }
```

Reply:
@Amejia481 if we have to guarantee ordering i.e. something should happen after an action has been applied we can use our `Middleware`: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt#L21

It already does that for `QueueDownloadAction`. The only other way would be a store observer / flow, but middleware is easier in this case.

Reply:
Thanks :)